autofit/non_linear/optimize/lbfgs.py

```diff
@@ -0,0 +1,251 @@
+            path_prefix=path_prefix,
+            unique_tag=unique_tag,
+            prior_passer=prior_passer,
+            initializer=initializer,
+            iterations_per_update=iterations_per_update,
+            session=session,
+            **kwargs
+        )
+
+        self.logger.debug("Creating LBFGS Search")
+
+    @property
+    def config_dict_options(self):
+
+        config_dict = copy.copy(self._class_config["options"]._dict)
+
+        for key, value in config_dict.items():
+            try:
+                config_dict[key] = self.kwargs[key]
+            except KeyError:
+                pass
+
+        return config_dict
+
+    def _fit(self, model: AbstractPriorModel, analysis : Analysis, log_likelihood_cap : float=None):
+        """
+        Fit a model using the scipy L-BFGS method and the Analysis class which contains the data and returns the log
+        likelihood from instances of the model, which the `NonLinearSearch` seeks to maximize.
+
+        Parameters
+        ----------
+        model
+            The model which generates instances for different points in parameter space.
+        analysis
+            Contains the data and the log likelihood function which fits an instance of the model to the data,
+            returning the log likelihood the `NonLinearSearch` maximizes.
+
+        Returns
+        -------
+        A result object comprising the Samples object that inclues the maximum log likelihood instance and full
+        chains used by the fit.
+        """
+        fitness_function = self.fitness_function_from_model_and_analysis(
+            model=model, analysis=analysis
+        )
+
+        if self.paths.is_object("x0"):
+
+            x0 = self.paths.load_object("x0")
+            total_iterations = self.paths.load_object("total_iterations")
+
+            self.logger.info("Existing PySwarms samples found, resuming non-linear search.")
+
+        else:
+
+            initial_unit_parameter_lists, initial_parameter_lists, initial_log_posterior_list = self.initializer.initial_samples_from_model(
+                total_points=1,
+                model=model,
+                fitness_function=fitness_function,
+            )
+
+            x0 = np.asarray(initial_parameter_lists[0])
+
+            total_iterations = 0
+
+            self.logger.info("No PySwarms samples found, beginning new non-linear search. ")
+
+        maxiter = self.config_dict_options["maxiter"]
+
+        if maxiter is None:
+            maxiter = 1e8
+
+        while total_iterations < maxiter:
+
+            iterations_remaining = maxiter - total_iterations
+
+            if self.iterations_per_update > iterations_remaining:
+                iterations = iterations_remaining
+            else:
+                iterations = self.iterations_per_update
+
+            if iterations > 0:
+
+                config_dict_options = self.config_dict_options
+                config_dict_options["maxiter"] = iterations

```

Comment:
Updating the config here looks weird and is probably not the best way to do things